projects/Mallard/src/screens/issue-screen.tsx

```diff
@@ -128,12 +138,55 @@ const IssueFronts = ({
     ListHeaderComponent,
     style,
 }: {
-    issue: Issue
+    issue: IssueWithFronts
     ListHeaderComponent?: ReactElement
     style?: StyleProp<ViewStyle>
 }) => {
     const { container, card } = useIssueScreenSize()
     const { width } = useDimensions()
+
+    const {
+        frontWithCards,
+        articleSpecs,
+    }: {
+        frontWithCards: (TFront & { cards: FlatCard[] })[]
+        articleSpecs: ArticleSpec[]
+    } = useMemo(
+        () =>
+            issue.fronts.reduce(
```

Comment:
Feels like we are doing a load of data transformation within a component? Which doesnt feel right.

Reply:
Might be worth checking the performance of that render function indeed. If that's performant enough then I think it can be okay, otherwise indeed we'd want to cache that transformation.